packages/hardhat-core/src/types/artifact.ts

```diff
@@ -0,0 +1,15 @@
+export interface Artifact {
```

Comment:
I think this module should algo include `DebugArtifact` and `BuildInfo`.

Reply:
And also, they should probably be named `SolidityArtifact`, etc.

Reply:
I think I changed my mind. `Artifact` should be generic, but maybe `DebugArtifact` and `BuildInfo` shouldn't.

What do you think, @fvictorio ?

Reply:
Yes, that's exactly why I thought about naming it `SolidityArtifact`. 

But everything works with a single artifact type now. Changing it to have multiple types would be too disruptive.

Reply:
Would it make sense to put it in the `types/solidity` module at least? We could alias `SolidityArtifact` to `Artifact` too.

Reply:
The issue is that, for compatibility reasons, vyper tends to use the same artifacts format.